### PR TITLE
Decompile unk_020910D8 -> hatch_egg_task

### PR DESCRIPTION
**Pull request by tgsm**

### CODE REVIEW

File: src/hatch_egg_task.c [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+    u32 state;
+    BOOL wantToNameNewMon;
+    NamingScreenArgs *namingScreenArgs;
+    UnkStruct_02091240 unkC;
+} HatchEggTaskData;
+
+FS_EXTERN_OVERLAY(OVY_95);
+extern BOOL ov95_021E6D70(OVY_MANAGER *, int *);
+extern BOOL ov95_021E6E88(OVY_MANAGER *, int *);
+extern BOOL ov95_021E6F0C(OVY_MANAGER *, int *);
```
Comment (PikalaxALT) [inline src/hatch_egg_task.c:19]: Create a header for these

File: src/hatch_egg_task.c [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+    .exec   = ov95_021E6E88,
+    .exit   = ov95_021E6F0C,
+    .ovy_id = FS_OVERLAY_ID(OVY_95),
+};
+
+static BOOL Task_HatchEggInParty(TaskManager *taskManager) {
+    HatchEggTaskData *data = TaskManager_GetEnvironment(taskManager);
+
+    switch (data->state) {
+    case 0: // Leave overworld
```
Comment (PikalaxALT) [inline src/hatch_egg_task.c:32]: Typedef an enum for this

File: src/hatch_egg_task.c [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+        CallApplicationAsTask(taskManager, &_02106048, data);
+        data->state++;
+        break;
+    case 2: { // Update egg status, add in mapsec for hatch location, update Pokedex if necessary
+        FieldSystem *fieldSystem = TaskManager_GetFieldSystem(taskManager);
+        Pokemon *pokemon         = data->unkC.mon;
+        PlayerProfile *profile   = Save_PlayerData_GetProfileAddr(FieldSystem_GetSaveData(fieldSystem));
+
+        u32 mapsec = MapHeader_GetMapSec(fieldSystem->location->mapId);
+        u32 isEgg  = 0;
```
Comment (PikalaxALT) [inline src/hatch_egg_task.c:47]: Should be a BOOL?

File: src/hatch_egg_task.c [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+};
+
+static BOOL Task_HatchEggInParty(TaskManager *taskManager) {
+    HatchEggTaskData *data = TaskManager_GetEnvironment(taskManager);
+
+    switch (data->state) {
+    case 0: // Leave overworld
+        sub_0206D328(data->unkC.mon, HEAP_ID_FIELD);
+        CallTask_LeaveOverworld(taskManager);
+        data->state++;
```
Comment (PikalaxALT) [inline src/hatch_egg_task.c:35]: Prefer prefix increment

**Review by PikalaxALT — CHANGES_REQUESTED** [review]

**Comment by adrienntindall** [comment]
build failed too btw

File: src/hatch_egg_task.c (hunk 3 above)
Comment (tgsm) [reply src/hatch_egg_task.c:47]: There was one spot in `sub_0206D328` where this value was set to 70. For now I'll make it a BOOL

File: src/hatch_egg_task.c (hunk 4 above)
Comment (tgsm) [reply src/hatch_egg_task.c:35]: Resolved in discord

**Review by PikalaxALT — APPROVED** [review]

File: src/hatch_egg_task.c [hunk 5]
```diff
@@ -0,0 +1,107 @@
+#include "hatch_egg_task.h"
+
+#include "get_egg.h"
+#include "naming_screen.h"
+#include "overlay_95.h"
+#include "trainer_memo.h"
+#include "unk_02055244.h"
+#include "unk_020552A4.h"
```
Comment (adrienntindall) [inline src/hatch_egg_task.c:8]: duplicate include

File: src/hatch_egg_task.c [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+
+typedef struct HatchEggTaskData {
+    HatchEggTaskState state;
+    BOOL wantToNameNewMon;
+    NamingScreenArgs *namingScreenArgs;
+    UnkStruct_02091240 unkC;
+} HatchEggTaskData;
+
+FS_EXTERN_OVERLAY(OVY_95);
+static const OVY_MGR_TEMPLATE _02106048 = {
```
Comment (adrienntindall) [inline src/hatch_egg_task.c:28]: Isn't this the hatch application? or is there still some ambiguity?

File: src/hatch_egg_task.c (hunk 5 above)
Comment (PikalaxALT) [reply src/hatch_egg_task.c:8]: 44 vs A4, this comment is invalid

Comment (adrienntindall) [reply src/hatch_egg_task.c:8]: it turns out I cannot read

File: src/hatch_egg_task.c (hunk 6 above)
Comment (tgsm) [reply src/hatch_egg_task.c:28]: Missed this one, whoops

File: src/hatch_egg_task.c [hunk 7]
```diff
@@ -25,7 +25,7 @@ typedef struct HatchEggTaskData {
 } HatchEggTaskData;
 
 FS_EXTERN_OVERLAY(OVY_95);
-static const OVY_MGR_TEMPLATE _02106048 = {
+static const OVY_MGR_TEMPLATE sApplication_HatchEgg = {
     .init   = ov95_021E6D70,
     .exec   = ov95_021E6E88,
     .exit   = ov95_021E6F0C,
```
Comment (PikalaxALT) [inline src/hatch_egg_task.c:31]: These funcs can be named now

**Review by PikalaxALT — CHANGES_REQUESTED** [review]

**Review by PikalaxALT — APPROVED** [review]

**Review by adrienntindall — APPROVED** [review]

**Pull request merged**